API.md

```diff
@@ -0,0 +1,552 @@
+| protocols | `String|Array<String>` | Strings that identifies the protocols (e.g '/ipfs/bitswap/1.1.0') |
+| [options] | `Object` | dial options |
+| [options.signal] | `AbortSignal` | abort signal |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Connection>` | Promise resolves with the [Connection](https://github.com/libp2p/js-interfaces/tree/master/src/connection) instance |
+
+#### Example
+
+```js
+// ...
+const conn = await libp2p.dialProtocol(remotePeerInfo, protocols)
+```
+
+### hangUp
+
+Closes an open connection with a peer, graciously.
+
+`hangUp(peer)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peer | [PeerInfo](https://github.com/libp2p/js-peer-info), [PeerId](https://github.com/libp2p/js-peer-id), [multiaddr](https://github.com/multiformats/js-multiaddr), `string` | peer to hang up |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<void>` | Promise resolves once connection closes |
+
+#### Example
+
+```js
+// ...
+await libp2p.hangUp(remotePeerInfo)
+```
+
+### handle
+
+Registers a given handler for the given protocols.
+
+`libp2p.handle(protocols, handler)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| protocols | `Array<String>|String` | protocols to register |
+| handler | `function({ connection:*, stream:*, protocol:string })` | handler to call |
+
+
+#### Example
+
+```js
+// ...
+const handler = ({ connection, stream, protocol }) => {
+  // use stream or connection according to the needs
+}
+
+libp2p.handle('/echo/1.0.0', handler)
+```
+
+### unhandle
+
+Unregisters all handlers with the given protocols
+
+`libp2p.unhandle(protocols)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| protocols | `Array<String>|String` | protocols to unregister |
+
+#### Example
+
+```js
+// ...
+libp2p.unhandle('/echo/1.0.0')
+```
+
+### peerRouting.findPeer
+
+Iterates over all peer routers in series to find the given peer.
+
+`libp2p.peerRouting.findPeer(peerId, options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peerId | [`PeerId`](https://github.com/libp2p/js-peer-id) | ID of the peer to find |
+| options | `Object` | operation options |
+| options.timeout | `number` | maximum time the query should run |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<PeerInfo>` | Peer info of a known peer |
+
+#### Example
+
+```js
+// ...
+const peerInfo = await libp2p.peerRouting.findPeer(peerId, options)
+```
+
+### contentRouting.findProviders
+
+Iterates over all content routers in series to find providers of the given key.
+Once a content router succeeds, the iteration will stop.
+
+`libp2p.contentRouting.findProviders(cid, options)`
```

Comment:
Please can this return an async iterable that yields `PeerInfo`.

Reply:
:+1: We haven't refactored this quite yet, should happen in the next day or two, and I intend to make all array based queries async iterable.

Reply:
Will be changed in the next commit :)